### PR TITLE
Working flashlight for hard hats

### PR DESCRIPTION
**Pull request by Macoron**
hard-hats
Continue #2362

Now hard hats have working flashlight. It use same component as original flashlight. 
Add verb to toggle off/on flashlight.

Moved hard hats sprites names to "off-on" prefix. Also add enabled light effect for some sprites.
Now ClothingEquippedPrefix will update players sprite when changed.

### CODE REVIEW

File: Content.Client/GameObjects/Components/Clothing/ClothingComponent.cs [hunk 1]
```diff
@@ -18,8 +22,28 @@ public class ClothingComponent : ItemComponent
...
-        public string ClothingEquippedPrefix { get; set; }
+        public string ClothingEquippedPrefix
+        {
+            get
+            {
+                return _clothingEquippedPrefix;
+            }
+            set
+            {
+                _clothingEquippedPrefix = value;
```
Comment (metalgearsloth) [inline Content.Client/GameObjects/Components/Clothing/ClothingComponent.cs:36]:
```suggestion
                if (_clothingEquippedPrefix == value)
                    return;

                _clothingEquippedPrefix = value;
```

File: Content.Client/GameObjects/Components/Clothing/ClothingComponent.cs [hunk 2]
```diff
@@ -18,8 +22,28 @@ public class ClothingComponent : ItemComponent
...
+        private string _clothingEquippedPrefix;
+
         [ViewVariables(VVAccess.ReadWrite)]
-        public string ClothingEquippedPrefix { get; set; }
+        public string ClothingEquippedPrefix
+        {
+            get
+            {
+                return _clothingEquippedPrefix;
+            }
```
Comment (metalgearsloth) [inline Content.Client/GameObjects/Components/Clothing/ClothingComponent.cs:33]:
```suggestion
            get => _clothingEquippedPrefix;
```

File: Content.Client/GameObjects/Components/HUD/Inventory/ClientInventoryComponent.cs [hunk 3]
```diff
@@ -214,5 +214,20 @@ public bool TryGetSlot(Slots slot, out IEntity item)
...
+                if (pair.Value == item)
+                {
+                    slots = pair.Key;
+                    return true;
+                }
+            }
+
+            slots = Slots.NONE;
+            return false;
+        }
```
Comment (metalgearsloth) [inline Content.Client/GameObjects/Components/HUD/Inventory/ClientInventoryComponent.cs:231]: Would be better to have the file marked as nullable and return a null slot instead IMO.

```suggestion
        public bool TryFindItemSlots(IEntity item, [NotNullWhen(true)] out Slots? slots)
        {
            slots = null;
        
            foreach (var (slot, entity) in _slots)
            {
                if (entity == item)
                {
                    slots = slot;
                    return true;
                }
            }

            return false;
        }
```

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

**Comment by Macoron** [comment]
Fixed merge conflicts with latest master and also added suggested changes.
Ready for review.

**Comment by SweptWasTaken** [comment]
Good PR :)

**Pull request merged**